File: solutions/go/hamming/2/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	
	var d int = 0;
	if len(a) != len(b) {	
		return 0, errors.New("bad arguments")
	}
	for i, _ := range a {
		if a[i] != b[i] {
			d++;
		}
	}
	return d, nil

}

File: solutions/r/hamming/2/hamming.R
# This is a stub function to take two strings
# and calculate the hamming distance
hamming <- function(strand1, strand2) {
	if (nchar(strand1) == nchar(strand2)) {
		return (sum((unlist(strsplit(strand1,"")) == unlist(strsplit(strand2,""))) == F))
	} else {
		stop("error")
	}
}
